lib/src/test/java/com/dynatrace/metric/util/DynatraceMetadataEnricherTest.java

```diff
@@ -0,0 +1,485 @@
+  }
+
+  @Test
+  public void testGetDynatraceMetadataFileContentInvalid() throws IOException {
+    List<String> inputs = Arrays.asList("=0", "", "a=", "\t\t", "=====", "    ", "   test   ");
+    List<String> expected = Arrays.asList("=0", "", "a=", "", "=====", "", "test");
+
+    StringReader reader = new StringReader(String.join("\n", inputs));
+    List<String> result = DynatraceMetadataEnricher.getDynatraceMetadataFileContents(reader);
+    assertEquals(expected, result);
+    assertNotSame(expected, result);
+  }
+
+  @Test
+  public void testGetDynatraceMetadataFileContentEmptyFile() throws IOException {
+    List<String> expected = new ArrayList<>();
+
+    List<String> result =
+        DynatraceMetadataEnricher.getDynatraceMetadataFileContents(new StringReader(""));
+    assertEquals(expected, result);
+    assertNotSame(expected, result);
+  }
+
+  @Test
+  public void testGetDynatraceMetadataFileContentPassNull() {
+    assertThrows(
+        IOException.class, () -> DynatraceMetadataEnricher.getDynatraceMetadataFileContents(null));
+  }
+
+  @Test
+  public void testGetMetadataFileContentWithRedirection_Valid() {
+    // this should not be used. If it is, it will not exist and throw an exception, breaking the
+    // test.
+    String nonExistentAlternativeFilename = generateNonExistentFilename();
+    List<String> expected = Arrays.asList("key1=value1", "key2=value2", "key3=value3");
+    List<String> results =
+        DynatraceMetadataEnricher.getMetadataFileContentWithRedirection(
+            "src/test/resources/indirection.properties", nonExistentAlternativeFilename);
+    assertEquals(expected, results);
+    assertNotSame(expected, results);
+  }
+
+  private String generateNonExistentFilename() {
+    File f;
+    Random r = new Random();
+    // generate random filenames until we find one that does not exist:
+    do {
+      byte[] array = new byte[7];
+      r.nextBytes(array);
+      String filename =
+          "src/test/resources/" + new String(array, StandardCharsets.UTF_8) + ".properties";
+
+      f = new File(filename);
+    } while (f.exists());
+    return f.getAbsolutePath();
+  }
+
+  @Test
+  public void
+      testGetMetadataFileContentWithRedirection_IndirectionFileDoesNotExistAlternativeDoesNotExist() {
+    String filename = generateNonExistentFilename();
+    String alternativeFilename = generateNonExistentFilename();
+
+    List<String> result =
+        DynatraceMetadataEnricher.getMetadataFileContentWithRedirection(
+            filename, alternativeFilename);
+    assertEquals(Collections.<String>emptyList(), result);
+  }
+
+  @Test
+  public void
+      testGetMetadataFileContentWithRedirection_IndirectionFileDoesNotExistAlternativeExists() {
+    String filename = generateNonExistentFilename();
+
+    List<String> result =
+        DynatraceMetadataEnricher.getMetadataFileContentWithRedirection(
+            filename, "src/test/resources/indirection_target.txt");
```

Comment:
In this case it's not the target of an indirection so the name is misleading. Please rename the file to `metadata_file.properties` or similar, this would fit in both cases (indirection target and alternative metadata file).